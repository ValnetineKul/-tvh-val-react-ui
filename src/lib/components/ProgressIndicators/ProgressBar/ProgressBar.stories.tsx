import { createMeta, createTemplate } from '../../../story-utils';
import ProgressBar from './ProgressBar';

const Template = createTemplate(ProgressBar);

export const Primitive = Template.bind({});
Primitive.args = {
  progress: 50,
};

Primitive.storyName = 'ProgressBar';

export const WithLabel = Template.bind({});
WithLabel.args = {
  progress: 50,
  endLabel: '500',
};

WithLabel.storyName = 'WithEndLabel';

export default createMeta({
  component: ProgressBar,
  title: 'Components/ProgressIndicators/ProgressBar',
  argTypes: {
    progress: {
      table: {
        type: {
          detail: 'percentage (0-100)',
        },
      },
    },
    endLabel: {
      table: {
        type: {
          detail: 'progress information label',
        },
      },
      control: {
        disable: true,
      },
    },
  },
});
