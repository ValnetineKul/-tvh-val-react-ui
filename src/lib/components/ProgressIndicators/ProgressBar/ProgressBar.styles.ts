import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles({ name: 'ProgressBar' })((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'baseline',
  },
  progressBar: {
    background: theme.color['bg/surface200/default'],
    borderRadius: 4,
    height: 8,
    flex: '1 1 auto',
    '& .MuiLinearProgress-barColorPrimary': {
      background: theme.color['bg/surfacePrimary/default'],
      borderRadius: 4,
    },
  },
  endLabel: {
    marginLeft: theme.layoutSpacing['spacing/300'],
  },
}));

export default useStyles;
