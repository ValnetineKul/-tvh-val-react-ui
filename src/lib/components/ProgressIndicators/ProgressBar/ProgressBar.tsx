import LinearProgress from '@mui/material/LinearProgress';

import type { FC } from 'react';
import React from 'react';
import Typography from '../../Typography';
import useStyles from './ProgressBar.styles';

export interface ProgressBarProps {
  progress: number;
  endLabel?: string | React.ReactElement;
  className?: string;
}

const ProgressBar: FC<ProgressBarProps> = ({ progress, endLabel, className }) => {
  const { classes, cx } = useStyles();

  const renderEndLabel = () => {
    return typeof endLabel === 'string' ? (
      <Typography variant="body400" className={classes.endLabel}>
        {endLabel}
      </Typography>
    ) : (
      <div className={classes.endLabel}>{endLabel}</div>
    );
  };

  return (
    <div className={cx(classes.root, className)}>
      <LinearProgress className={cx(classes.progressBar)} variant="determinate" value={progress} />
      {endLabel && renderEndLabel()}
    </div>
  );
};
export default ProgressBar;
