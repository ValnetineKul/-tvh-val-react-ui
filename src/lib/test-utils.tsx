import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
