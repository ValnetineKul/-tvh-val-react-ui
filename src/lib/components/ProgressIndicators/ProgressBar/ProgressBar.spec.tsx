import React from 'react';
import { render } from '../../../test-utils';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  it('Should render correctly', () => {
    const { container } = render(<ProgressBar progress={50} endLabel="50%" />);
    expect(container.firstChild).toMatchSnapshot('progressbar');
  });
});
