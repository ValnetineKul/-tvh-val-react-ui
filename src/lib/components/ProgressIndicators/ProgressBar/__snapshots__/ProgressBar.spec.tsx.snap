// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Should render correctly: progressbar 1`] = `
<div
  class="tss-1kejtv2-ProgressBar-root"
>
  <span
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="50"
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate tss-1r1iwg9-ProgressBar-progressBar mui-eglki6-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate mui-5xe99f-MuiLinearProgress-bar1"
      style="transform: translateX(-50%);"
    />
  </span>
  <p
    class="tss-0-Typography-weightRegular-ref tss-q925wq-Typography-root-Typography-weightRegular-ProgressBar-endLabel-Typography-body400"
  >
    50%
  </p>
</div>
`;
